query/compile.go

```diff
@@ -550,6 +552,61 @@ func (c *compiledField) compileMovingAverage(args []influxql.Expr) error {
 	}
 }
 
+func (c *compiledField) compileExponentialMovingAverage(args []influxql.Expr) error {
+	if got := len(args) - 1; got < 1 || got > 3 {
+		return fmt.Errorf("invalid number of arguments for exponential_moving_average, expected at least 1 but no more than 3, got %d", got)
+	}
+
+	switch arg1 := args[1].(type) {
+	case *influxql.IntegerLiteral:
+		if arg1.Val < 1 {
+			return fmt.Errorf("exponential_moving_average period must be greater than or equal to 1")
+		}
+	default:
+		return fmt.Errorf("exponential_moving_average period must be an integer")
+	}
+
+	if len(args) >= 3 {
+		switch arg2 := args[2].(type) {
+		case *influxql.IntegerLiteral:
+			if arg2.Val < 0 && arg2.Val != -1 {
+				return fmt.Errorf("exponential_moving_average hold period must be greater than or equal to 0")
+			}
+		default:
+			return fmt.Errorf("exponential_moving_average hold period must be an integer")
+		}
+	}
+
+	if len(args) >= 4 {
```

Comment:
This isn't possible. The portion above expects the number of arguments to be 3 or less.